recipes/stlab/all/conanfile.py

```diff
@@ -11,48 +11,168 @@ class Stlab(ConanFile):
+            raise ConanInvalidConfiguration("{}/{} task_system=windows is not supported on {}. Try using task_system=auto".format(self.name, self.version, self.settings.os))
+
+    def _validate_task_system_emscripten(self):
+        if self.settings.os != "Emscripten":
+            raise ConanInvalidConfiguration("{}/{} task_system=emscripten is not supported on {}. Try using task_system=auto".format(self.name, self.version, self.settings.os))
+
+    def _validate_task_system(self):
+        if self.options.task_system == "libdispatch":
+            self._validate_task_system_libdispatch()
+        elif self.options.task_system == "windows":
+            self._validate_task_system_windows()
+        elif self.options.task_system == "emscripten":
+            self._validate_task_system_emscripten()
+
+    def _validate_boost_components(self):
+        if self.settings.os != "Macos": return
+        if self.settings.compiler != "apple-clang": return
+        if Version(self.settings.compiler.version) >= "12": return
+        if self.options.boost_optional and self.options.boost_variant: return
+        #
+        # On Apple we have to force the usage of boost.variant, because Apple's implementation of C++17 
+        # is not complete.
+        #
+        msg = "Apple-Clang versions less than 12 do not correctly support std::optional or std::variant, so we will use boost::optional and boost::variant instead. "
+        if not self.options.boost_optional and not self.options.boost_variant:
+            msg += "Try -o boost_optional=True -o boost_variant=True"
+        elif not self.options.boost_optional:
+            msg += "Try -o boost_optional=True."
+        else:
+            msg += "Try -o boost_variant=True."
+
+        raise ConanInvalidConfiguration(msg)
+
+    def validate(self):
         if self.settings.compiler.get_safe("cppstd"):
             tools.check_min_cppstd(self, '17')
 
         if self.settings.compiler == "gcc" and Version(self.settings.compiler.version) < "9":
-            raise ConanInvalidConfiguration("Need gcc >= 9")
+            raise ConanInvalidConfiguration("Need GCC >= 9")
 
         if self.settings.compiler == "clang" and Version(self.settings.compiler.version) < "8":
-            raise ConanInvalidConfiguration("Need clang >= 8")
+            raise ConanInvalidConfiguration("Need Clang >= 8")
 
         if self.settings.compiler == "Visual Studio" and Version(self.settings.compiler.version) < "15.8":
             raise ConanInvalidConfiguration("Need Visual Studio >= 2017 15.8 (MSVC 19.15)")
 
+        self._validate_task_system()
+        self._validate_boost_components()
+
+    def config_options(self):
+        if self.options.task_system == "auto":
+            self.options.task_system = self._default_task_system()
+
+    def configure(self):
+        self.output.info("Stlab Task System: {}.".format(self.options.task_system))
+
```

Comment:
```suggestion
    def configure(self):
        if self.options.task_system == "auto":
            self.options.task_system = self._default_task_system()
        self.output.info("Stlab Task System: {}.".format(self.options.task_system))

```
config_options can not evaluate options provided by the user, it will adopt default_options values ever. It only works for settings.

Reply:
Thanks!

Reply:
Hmmm that's exactly how I expected the `"auto"` to work...

...until the user _explicitly_ sets the option to `"auto"` and the whole system breaks

Good catch!